lib/galaxy/objectstore/irods.py

```diff
@@ -235,8 +209,40 @@ def __init__(self, config, config_dict):
             raise Exception(IRODS_IMPORT_MESSAGE)
 
         self.home = "/" + self.zone + "/home/" + self.username
+        self._initialize()
         log.debug("irods __init__ %s", reload_timer)
 
+    def shutdown(self):
+        # This call will cleanup all the connections in the connection pool
+        # OSError sometimes happens on GitHub Actions, after the test has successfully completed. Ignore it if it happens.
+        try:
+            self.session.cleanup()
+        except OSError:
+            pass
+
+    def _initialize(self):
+        if irods is None:
+            raise Exception(IRODS_IMPORT_MESSAGE)
+
+        self.session = self._configure_connection(host=self.host, port=self.port, user=self.username, password=self.password, zone=self.zone, timeout=self.timeout, poolsize=self.poolsize)
+
+    def _configure_connection(self, host='localhost', port='1247', user='rods', password='rods', zone='tempZone', timeout=30, poolsize=1):
+        session = iRODSSession(host=host, port=port, user=user, password=password, zone=zone)
+        # Set connection timeout
+        session.connection_timeout = timeout
+        # Throws NetworkException if connection fails
+        try:
+            # We will create as many conections as poolsize.
+            # After creating the connection, release  it
+            # so it goes into the idle connection queue
+            for idx in range(poolsize):
+                conn = session.pool.get_connection()
+                session.pool.release_connection(conn)
+        except NetworkException as e:
+            log.error('Could not create iRODS session: ' + str(e))
+            raise
+        return session
```

Comment:
You can remove all of this, that's just a remnant of one connection per irods interaction. We really just need to setup one connection when setting up the objectstore and one for tearing down. Pool management is done by python-irodsclient.

Reply:
Ok, removed the code and simplified it.